### PR TITLE
MINIFI-170 Change FlowStatusReport to be JSON

### PR DESCRIPTION
**Pull request by apsaltis**
The FlowStatus report were in a format that is close to but not JSON.

FlowStatusReport{controllerServiceStatusList=null, processorStatusList=[{name='GenerateFlowFile', processorHealth={runStatus='Running', hasBulletins=false, validationErrorList=[]}, processorStats={activeThreads=0, flowfilesReceived=0, bytesRead=0, bytesWritten=1024000, flowfilesSent=0, invocations=100, processingNanos=106080516}, bulletinList=[]}], connectionStatusList=null, remoteProcessGroupStatusList=null, instanceStatus=null, systemDiagnosticsStatus=null, reportingTaskStatusList=null, errorsGeneratingReport=[]}

This makes it hard to process. The changes in this P/R result in a FlowStatus report that is valid JSON, thus enabling standard tools to be used to process the reports.

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

File: minifi-commons/minifi-utils/pom.xml [hunk 1]
```diff
@@ -15,6 +15,12 @@
 -->
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
+    <dependencies>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
```
Comment (apsaltis) [inline minifi-commons/minifi-utils/pom.xml]: Certainly.

File: minifi-commons/minifi-utils/src/main/java/org/apache/nifi/minifi/commons/status/FlowStatusReport.java [hunk 2]
```diff
@@ -111,14 +116,20 @@ public boolean equals(Object o) {
 
         FlowStatusReport that = (FlowStatusReport) o;
 
-        if (getControllerServiceStatusList() != null ? !getControllerServiceStatusList().equals(that.getControllerServiceStatusList()) : that.getControllerServiceStatusList() != null) return false;
```
Comment (apsaltis) [inline minifi-commons/minifi-utils/src/main/java/org/apache/nifi/minifi/commons/status/FlowStatusReport.java]: certainly

**Comment by apiri** [comment]
Updates look good.  Thanks for patching those up.  Will merge this in.

**Pull request closed without merging**